### DISCUSSION TITLE
Bad train loss and good generate image result when use train_dreambooth_lora_flux script.py

### DISCUSSION THREAD

**Discussion by vagitablebirdcode** [huggingface/diffusers#12088] [Q&A] [answered]
<img width="2384" height="1078" alt="图片" src="https://github.com/user-attachments/assets/206fcd49-1362-4ac9-89df-26fc5e2194ff" />
<img width="1521" height="1010" alt="图片" src="https://github.com/user-attachments/assets/62ecd96c-c2a2-4b62-b44a-c41b06ef4144" />
Hi everyone,

As the title says, I'm trying to fine-tune using the Flux script with the dog example dataset. Surprisingly, even though the training loss fluctuates heavily, the fine-tuned results are quite good!

This confuses me a bit — I was expecting the training loss to decrease smoothly (ideally in an exponential-like curve), but instead it's very noisy. Still, the final outputs look fine. So I'm not sure how meaningful the training loss actually is in this context.

For comparison, I tried fine-tuning on other platforms like ModelScope, and their results are also good — but their training loss curves (e.g., see Figure 2 above) look much smoother and more in line with intuition.

I'd love to understand why our fine-tuning script results in such unstable training loss (despite good performance), and if there's anything we can do to make the loss values more interpretable or stable like on other platforms.

Thanks in advance!

**Comment by asomoza** [comment, marked as answer]
Hi, there's a difference between showing the steps and the epochs, depending on how big your dataset, this makes a big difference, in my understanding the way we show the loss is the correct one (and it is raw data), since the training depends on noise, we get some variations but if you display the epochs you will probably see a much better curve, overall the values should still go down.

Do you have access to the modelscope loss function? Do they smooth the curve? Are you also training a dreambooth lora with the same hyperparameters?

I think [this article](https://medium.com/@damian0815/fine-tuning-stable-diffusionwith-validation-3fe1395ab8c3) provides helpful insights on the loss curve.

It seems to me they're doing something similar to what that article points in the EveryDream2 trainer.

**Comment by vagitablebirdcode** [comment]
<img width="1770" height="1134" alt="图片" src="https://github.com/user-attachments/assets/06399a7e-b08d-45e2-8a32-0bb1ac0366a8" />
Thanks a lot for the suggestions and reference! After applying a smoothing window to the loss curve, I found that the overall trend is indeed decreasing. With the smoothed plot, the loss curve is now much more interpretable and definitely useful for monitoring.
